src/Microsoft.Kusto.ServiceLayer/Connection/ConnectionService.cs

```diff
@@ -845,15 +829,13 @@ public void RegisterOnDisconnectTask(OnDisconnectHandler activity)
         /// <param name="connectParams"></param>
         /// <param name="requestContext"></param>
         /// <returns></returns>
-        protected async Task HandleConnectRequest(
-            ConnectParams connectParams,
-            RequestContext<bool> requestContext)
+        private async Task HandleConnectRequest(ConnectParams connectParams, RequestContext<bool> requestContext)
         {
             Logger.Write(TraceEventType.Verbose, "HandleConnectRequest");
 
             try
             {
-                RunConnectRequestHandlerTask(connectParams);
+                Parallel.Invoke(async () => await RunConnectRequestHandlerTask(connectParams));
```

Comment:
Moved Task.Run up the stack to HandleConnectRequest instead of RunConnectRequestHandlerTask